src/main/java/apoc/bolt/Bolt.java

```diff
@@ -0,0 +1,160 @@
+        boolean addStatistics = (boolean) config.getOrDefault("statistics", false);
+        UriResolver uri = new UriResolver(url, "bolt");
+        uri.initialize();
+        Driver driver = null;
+        Session session = null;
+        try {
+            driver = GraphDatabase.driver(uri.getUrlDriver(), AuthTokens.basic(uri.getUser(), uri.getPassword()));
+            session = driver.session();
+            if (addStatistics)
+                return Stream.of(new RowResult(toMap(runStatement(statement, session, params).summary().counters())));
+
+            return getRowResultStream(virtual, session, params, statement);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            session.close();
+            driver.close();
+        }
+    }
+
+    @Procedure()
+    @Description("apoc.bolt.execute(url-or-key, statement, params, config) - access to other databases via bolt for read")
+    public Stream<RowResult> execute(@Name("url") String url, @Name("statement") String statement, @Name(value = "params", defaultValue = "{}") Map<String, Object> params, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) throws URISyntaxException {
+        Map<String, Object> configuration =  new HashMap<>(config);
+        configuration.remove("virtual");
+        return load(url, statement, params, configuration);
+    }
+
+    private StatementResult runStatement(@Name("statement") String statement, Session session, Map<String, Object> finalParams) {
+        return session.writeTransaction((Transaction tx) -> tx.run(statement, finalParams));
+    }
+
+    private Stream<RowResult> getRowResultStream(boolean virtual, Session session, Map<String, Object> params, String statement) {
+        return runStatement(statement, session, params).list().stream()
+                .map(record -> record.fields())
+                .map(fields -> {
+                    Map<String, Object> map = new HashMap<>();
+                    for (int i = 0; i < fields.size(); i++) {
+                        String key = fields.get(i).key();
+                        Object value = fields.get(i).value().asObject();
+                        if (value instanceof Path) map.putAll(toPath(value, virtual, key));
+                        else if (value instanceof Node) map.putAll(toNode(value, key, virtual));
+                        else if (value instanceof Relationship) map.putAll(toRelationship(value, map, key, virtual));
+                        else map.put(key, value);
+                    }
+                    return map;
+                })
+                .map(resultMap -> new RowResult(resultMap));
+    }
+
+    private Map<String, Object> toNode(Object value, String key, boolean virtual) {
+        Value internalValue = ((InternalEntity) value).asValue();
+        Node node = internalValue.asNode();
+        if (virtual) {
+            VirtualNode virtualNode = new VirtualNode(node.id(), db);
```

Comment:
VirtualNode has also a constructor that should take id,db,labels(as strings) and properties as map